File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/view/adapter/ImagesAdapter.java
package technology.mainthread.apps.gatekeeper.view.adapter;

import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.squareup.picasso.Picasso;

import java.util.List;

import technology.mainthread.apps.gatekeeper.R;
import technology.mainthread.apps.gatekeeper.databinding.ItemImageBinding;

public class ImagesAdapter extends RecyclerView.Adapter<ImagesAdapter.ViewHolder> {

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemImageBinding binding;

        public ViewHolder(View view) {
            super(view);
            binding = ItemImageBinding.bind(view);
        }

        public void bindTo(Picasso picasso, Uri imageUri) {
            picasso.load(imageUri).into(binding.image);
        }
    }

    private Picasso picasso;
    private final List<Uri> images;

    public ImagesAdapter(Picasso picasso, List<Uri> images) {
        this.picasso = picasso;
        this.images = images;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_image, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.bindTo(picasso, images.get(position));
    }

    @Override
    public int getItemCount() {
        return images.size();
    }

}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/view/adapter/ImagesAdapter.kt
package technology.mainthread.apps.gatekeeper.view.adapter

import android.net.Uri
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.squareup.picasso.Picasso

import technology.mainthread.apps.gatekeeper.R
import technology.mainthread.apps.gatekeeper.databinding.ItemImageBinding

class ImagesAdapter(private val picasso: Picasso, private val images: List<Uri>) : RecyclerView.Adapter<ImagesAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        private val binding: ItemImageBinding = ItemImageBinding.bind(view)

        fun bindTo(picasso: Picasso, imageUri: Uri) {
            picasso.load(imageUri).into(binding.image)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_image, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindTo(picasso, images[position])
    }

    override fun getItemCount(): Int {
        return images.size
    }

}
